File: Revision_classes/06.Dv, LVP,PassByValue/Test6.java
/*
class Test6 
{
	static int a;
	static void m1(int x) {
		a = x;
		System.out.println(a);
	}
	
	public static void main(String[] args) 
	{
		System.out.println(a); 
		m1(50);
		System.out.println(a);
	}
}
*/

/*
class Test6
{
	static int a;
	static void m1(int a) {
		a = a;
		System.out.println(a); //50
	}
	
	public static void main(String[] args) 
	{
		System.out.println(a);
		m1(50);
		System.out.println(a); // 0
	}
}
*/
/*
class Test6 
{
	static int a;
	static void m1(int a) {
		Test6.a = a;
		System.out.println(a);
	}
	
	public static void main(String[] args) 
	{
		System.out.println(a);
		m1(50);
		System.out.println(a);
	}
}
*/

class Test6
{
	static int a;
	static void m1(int a) {
		 a = Test6.a ;
		System.out.println(a);
	}
	
	public static void main(String[] args) 
	{
		System.out.println(a);
		m1(50);
		System.out.println(a);
	}
}

File: React/app7/src/Test6.js
export default function Test6() {
    const doTask1 = (arg1) => {
        return new Promise((resolve, reject) => {
            //do some logic
            console.log('from doTask1:' + arg1)
            return resolve(arg1 + 200);
           //reject('error in task1');
        })
    }
    const doTask2 = (arg1) => {
        return new Promise((resolve, reject) => {
            //do some logic
            console.log('from doTask2:' + arg1)
            return resolve(arg1 + 200);
        })
    }
    const doTask3 = (arg1) => {
        return new Promise((resolve, reject) => {
            //do some logic
            console.log('from doTask3:' + arg1)
            return resolve(arg1 + 300);
        })
    }
    const test = () => {
        doTask1(10)
        .then(result1 => doTask2(result1))
        .then(result2 => doTask3(result2))
        .then(finalResult => console.log('Final Result:', finalResult))
        .catch(err => console.error(err));
    }
    return (
        <>
            <button type="button" onClick={test}>test</button>
        </>
    );
}